java-checks-testkit/src/main/java/org/sonar/java/checks/verifier/PomCheckVerifier.java

```diff
@@ -104,7 +105,12 @@ public void reportIssue(PomCheck check, LocatedTree tree, String message) {
     public void reportIssue(PomCheck check, int line, String message, List<Location> secondary) {
       AnalyzerMessage analyzerMessage = new AnalyzerMessage(check, getFile(), line, message, 0);
       for (Location location : secondary) {
-        AnalyzerMessage secondaryLocation = new AnalyzerMessage(check, getFile(), location.tree.startLocation().line(), location.msg, 0);
+        AnalyzerMessage secondaryLocation;
+        if (location.onLine()) {
+          secondaryLocation = new AnalyzerMessage(check, getFile(), location.line, location.msg, 0);
```

Comment:
You can move the logic of getting the line from Location into the Location class and that way no need to duplicate constructor call here